modules/ocl-loader/src/main/java/org/eclipse/xpanse/modules/ocl/loader/data/models/enums/UserDataType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.ocl.loader.data.models.enums;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * user data types for UserData.
+ */
+public enum UserDataType {
+    SHELL,
+    POWERSHELL;
+
+    private static Map<String, UserDataType> enumsMap = new HashMap<String, UserDataType>(
+            2);
+
+    static {

```

Comment:
Its better if we can avoid these maps and define enums as below.
SHELL("shell"),
POWERSHELL("powershell");

the full class will look something like this (have not tested it). This gives a better readability of enum values.
```
public enum UserDataType {

    SHELL("shell"),
    POWEpublic enum UserDataType {

    SHELL("shell"),
    POWERSHELL("powershell");

    private final String scriptType;
RSHELL("powershell");

    private final String scriptType;

    UserDataType(String scriptType) {
        this.scriptType = scriptType;
    }

    /**
     * For UserDataType serialize.
     */
    @JsonCreator
    public UserDataType getByValue(String scriptType) {
        for (UserDataType userDataType : values()) {
            if (userDataType.scriptType.equals(scriptType)) {
                return userDataType;
            }
        }
        return null;
    }

    /**
     * For UserDataType deserialize.
     */
    @JsonValue
    public String toValue() {
        return this.scriptType;
    }
}

```